src/core/Entry.cpp

```diff
@@ -285,6 +287,77 @@ const EntryAttachments* Entry::attachments() const
     return m_attachments;
 }
 
+bool Entry::hasTotp() const
+{
+    return m_attributes->hasKey("TOTP Seed") || m_attributes->hasKey("otp");
+}
+
+QString Entry::totp() const
+{
+    if (hasTotp()) {
+        QString seed = totpSeed();
+        quint64 time = QDateTime::currentDateTime().toTime_t();
+        QString output = QTotp::generateTotp(seed.toLatin1(), time, m_data.totpDigits, m_data.totpStep);
+
+        return QString(output);
+    } else {
+        return QString("");
+    }
+}
+
+void Entry::setTotp(const QString& seed, quint8& step, quint8& digits)
+{
+    if (step == 0) {
+        step = 30;
+    }
+
+    if (digits == 0) {
+        digits = 6;
+    }
+
+    if (m_attributes->hasKey("otp")) {
+        m_attributes->set("otp", QString("key=%1&step=%2&size=%3").arg(seed).arg(step).arg(digits), true);
+    } else {
+        m_attributes->set("TOTP Seed", seed, true);
+        m_attributes->set("TOTP Settings", QString("%1;%2").arg(step).arg(digits));
+    }
+}
+
+QString Entry::totpSeed() const
+{
+    QString secret = "";
+
+    if (m_attributes->hasKey("otp")) {
+        secret = m_attributes->value("otp");
+    } else {
+        secret = m_attributes->value("TOTP Seed");
+    }
+
+    m_data.totpDigits = 6;
+    m_data.totpStep = 30;
```

Comment:
@weslly the default steps and digits are defined 3 times in this file. Maybe we could extract them as constants and place them somewhere in the totp files.